docs/usage/configuration-options.md

```diff
@@ -1080,6 +1080,32 @@ Warning: `pipenv` support is currently in beta, so it is not enabled by default.
 - `yarnDedupeFewer`: Run `yarn-deduplicate --strategy fewer` after `yarn.lock` updates
 - `yarnDedupeHighest`: Run `yarn-deduplicate --strategy highest` after `yarn.lock` updates
 
+## postUpgradeFiles
+
+A list of [minimatch](https://www.npmjs.com/package/minimatch) glob-style pattern matchers that determine which files will be added to the final commit after `postUpgradeTasks` have been run.
+
+e.g.
+
+```json
+{
+  "postUpgradeFiles": ["yarn.lock", "*.js"]
+}
+```
+
+## postUpgradeTasks
+
+A list of commands that are executed after Renovate has updated a dependency but before the commit it made. This is useful for running custom scripts or plugins. This is only available on Renovate instances that have a `trustLevel` of 'high'.
+
+Each command must match at least one of the patterns defined in `allowedPostUpgradeTasks` in order to be executed. If the list of allowed tasks is empty then no tasks will be executed.
+
+e.g.
+
+```json
+{
+  "postUpgradeTasks": ["tslint --fix"]
+}
+```
+
```

Comment:
Can we join this to a single options object?

Reply:
In theory it should not be possible to have files or tasks without the other, right @daekene? If so then ideally the config should be as @ViceIce suggests where the semantics themselves stop someone making the mistake of configuring one and not the other. I'm not sure of exactly how to do that either.

Reply:
@rarkins That's correct, it shouldn't be possible to have tasks without specifying files to include (the intention being that tasks should only be run to include additional files in the commit being created) so the config should tie these two things together.

I've merged these properties into `postUpdateTasks` object, with `commands` and `fileFilters` as two properties of that. 

`allowedPostUpdateCommands` is still separate as that's an administrative setting